File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.java
class Solution {
    public static String reverse(String s){
        StringBuilder sb = new StringBuilder();
        for(int i = s.length() - 1; i >= 0; i--){
            sb.append(s.charAt(i));
        }
        return sb.toString();
    }
    
    public static int makeBin(int n){
        StringBuilder sb = new StringBuilder();
        int result = 0;
        
        while(n != 0){
            sb.append(String.valueOf(n%3));
            n /= 3;
        }
        
        int mul = 1;
        String tmp = reverse(sb.toString());
        for(char c : tmp.toCharArray()){
            result += (c - '0') * mul;
            mul *= 3;
        }
        return result;
    }
    
    public int solution(int n) {
        return makeBin(n);
    }
}

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.py
def makeBin(a):
    result = ""
    while(a != 0) :
        result += str(a % 3)
        a //= 3
    return result

def solution(n):
    answer = 0
    bin3 = makeBin(n)
    rev = [ord(i) - ord('0') for i in bin3]
    rev.reverse()
    
    mul = 1
    for i in rev:
        answer += mul * i
        mul *= 3
        
    return answer